File: frontend/src/routes/_shared/materialReviewReport.ts
import { buildApiUrl, requestJson } from './api';
import type { Language } from './i18n';

export type MaterialReviewReportStatus =
	| 'generating'
	| 'ready'
	| 'ready_with_warnings'
	| 'failed';

export type MaterialReviewReadiness = 'strong' | 'usable' | 'preliminary' | 'insufficient';

export type MaterialReviewReport = {
	report_id: string;
	collection_id: string;
	material_id: string;
	status: MaterialReviewReportStatus;
	message: string;
	title: string | null;
	language: string;
	report_type: string;
	include_appendix: boolean;
	readiness: MaterialReviewReadiness;
	readiness_reason: string;
	data_version: string;
	warnings: string[];
	created_at: string;
	updated_at: string;
	generated_at: string | null;
	pdf_url: string | null;
	markdown_url: string | null;
};

type CreateMaterialReviewReportOptions = {
	language?: Language;
	report_type?: 'review_draft';
	include_appendix?: boolean;
	force_regenerate?: boolean;
};

function materialReviewReportPath(collectionId: string, materialId: string, suffix = '') {
	return `/collections/${encodeURIComponent(collectionId)}/materials/${encodeURIComponent(
		materialId
	)}/review-report${suffix}`;
}

export async function fetchMaterialReviewReport(collectionId: string, materialId: string) {
	return (await requestJson(materialReviewReportPath(collectionId, materialId), {
		method: 'GET'
	})) as MaterialReviewReport;
}

export async function createMaterialReviewReport(
	collectionId: string,
	materialId: string,
	options: CreateMaterialReviewReportOptions = {}
) {
	return (await requestJson(materialReviewReportPath(collectionId, materialId), {
		method: 'POST',
		body: JSON.stringify({
			language: options.language ?? 'zh',
			report_type: options.report_type ?? 'review_draft',
			include_appendix: options.include_appendix ?? true,
			force_regenerate: options.force_regenerate ?? false
		})
	})) as MaterialReviewReport;
}

export function buildMaterialReviewMarkdownUrl(collectionId: string, materialId: string) {
	return buildApiUrl(materialReviewReportPath(collectionId, materialId, '.md'));
}

export function buildMaterialReviewPdfUrl(collectionId: string, materialId: string) {
	return buildApiUrl(materialReviewReportPath(collectionId, materialId, '.pdf'));
}

File: backend/controllers/schemas/derived/material_review_report.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


MaterialReviewReportStatus = Literal[
    "generating",
    "ready",
    "ready_with_warnings",
    "failed",
]
MaterialReviewReadiness = Literal[
    "strong",
    "usable",
    "preliminary",
    "insufficient",
]


class MaterialReviewReportRequest(BaseModel):
    """Request body for generating an AI-assisted material review draft."""

    language: Literal["zh", "en"] = Field(default="zh", description="报告语言")
    report_type: Literal["review_draft"] = Field(
        default="review_draft",
        description="报告类型；MVP 仅支持综述论文草稿",
    )
    include_appendix: bool = Field(default=True, description="是否包含证据附录")
    force_regenerate: bool = Field(default=False, description="是否强制重新生成")


class MaterialReviewReportResponse(BaseModel):
    """Status and artifact links for one material review draft."""

    report_id: str = Field(..., description="报告 ID")
    collection_id: str = Field(..., description="集合 ID")
    material_id: str = Field(..., description="材料 ID")
    status: MaterialReviewReportStatus = Field(..., description="报告生成状态")
    stage: str | None = Field(default=None, description="报告生成流水线阶段")
    message: str = Field(..., description="状态说明")
    title: str | None = Field(default=None, description="报告标题")
    language: str = Field(default="zh", description="报告语言")
    report_type: str = Field(default="review_draft", description="报告类型")
    include_appendix: bool = Field(default=True, description="是否包含附录")
    readiness: MaterialReviewReadiness = Field(..., description="综述生成就绪度")
    readiness_reason: str = Field(..., description="就绪度理由")
    data_version: str = Field(..., description="材料档案上下文版本")
    warnings: list[str] = Field(default_factory=list, description="证据引用或生成警告")
    created_at: str = Field(..., description="创建时间")
    updated_at: str = Field(..., description="更新时间")
    generated_at: str | None = Field(default=None, description="生成完成时间")
    pdf_url: str | None = Field(default=None, description="PDF 下载链接")
    markdown_url: str | None = Field(default=None, description="Markdown 预览链接")
